types/akamai-edgeworkers/index.d.ts

```diff
@@ -1359,3 +1359,125 @@ declare module "crypto" {
+         * When the HtmlRewritingStream calls the handler, it passes an Element object as an argument.
+         */
+        onElement(selector: string, handler: (element: Element) => void): void;
+    }
+
+    interface GenericHtmlRewritingStream {
+        readonly readable: ReadableStream;
+        readonly writable: WritableStream;
+    }
+
+    /**
+     * The Element object is an argument to the handler registered with onElement(),
+     * the handler calls functions on the Element to modify the output stream.
+     */
+    interface Element {
+        /**
+         * Insert new content immediately after the end tag of the matched element.
+         * @param tag is the new text to insert.
+         * @param trailing_opt When the value is true, elements that are missing a close tag will have one inserted.
+         */
+        after(tag: string, trailing_opt?: TrailingOpt): void;
+
+        /**
+         * Insert content right before the end tag of the element.
+         * @param tag is the new text to insert.
+         * @param trailing_opt When the value is true, elements that are missing a close tag will have one inserted.
+         */
+        append(tag: string, trailing_opt?: TrailingOpt): void;
+
+        /**
+         * Insert new content immediately before the start tag of the matched element.
+         * @param tag is the new text to insert.
+         */
+        before(tag: string): void;
+
+        /**
+         * Read the value of a given attribute name on the tag or undefined if it doesn’t exist.
+         * @param tag is the case-insensitive name of the attribute.
+         */
+        getAttribute(tag: string): string | undefined;
+
+        /**
+         * Insert content right after the start tag of the element.
+         * @param tag is the new text to insert.
+         */
+        prepend(tag: string): void;
+
+        /**
+         * Removes the attribute if exists.
+         * @param tag is the case-insensitive name of the attribute. If an attribute was removed, it is returned.
+         */
+        removeAttribute(tag: string): string | undefined;
+
+        /**
+         * Remove the children of the current element and insert content in place of them.
+         * @param tag is the text to replace.
+         * @param trailing_opt When the value is true, elements that are missing a close tag will have one inserted.
+         */
+        replaceChildren(tag: string, trailing_opt?: TrailingOpt): void;
+
+        /**
+         * Remove the current element and its children, and insert the passed content in its place.
+         * @param tag is the text to replace.
+         */
+        replaceWith(tag: string): void;
+
+        /**
+         * Set an attribute to a provided value, creating the attribute if it doesn’t exist.
+         * @param name is case-insensitive. Any non-string value will be converted into a string, e.g., null will be converted to "null".
```

Comment:
Happily this is _not_ the case: we require the caller to pass a string into the function.